.github/workflows/pytest-workflow.yml

```diff
@@ -0,0 +1,342 @@
+          - profile: conda
+            tags: parabricks/haplotypecaller
+          - profile: conda
+            tags: parabricks/indexgvcf
+          - profile: conda
+            tags: parabricks/mutectcaller
+          - profile: conda
+            tags: picard/collecthsmetrics
+          - profile: conda
+            tags: picard/collectwgsmetrics
+          - profile: conda
+            tags: sentieon/applyvarcal
+          - profile: conda
+            tags: sentieon/datametrics
+          - profile: conda
+            tags: sentieon/dnamodelapply
+          - profile: conda
+            tags: sentieon/dnascope
+          - profile: conda
+            tags: sentieon/readwriter
+          - profile: conda
+            tags: sentieon/tnfilter
+          - profile: conda
+            tags: sentieon/tnhaplotyper2
+          - profile: conda
+            tags: sentieon/tnscope
+          - profile: conda
+            tags: sentieon/varcal
+          - profile: conda
+            tags: sentieon/wgsmetrics
+          - profile: conda
+            tags: subworkflows/bam_qc_picard
+          - profile: conda
+            tags: subworkflows/bcl_demultiplex
+          - profile: conda
+            tags: subworkflows/fasta_clean_fcs
+          - profile: conda
+            tags: svanalyzer/svbenchmark
+          - profile: conda
+            tags: universc
+          - profile: singularity
+            tags: universc
+          - profile: conda
+            tags: vt/decompose
+    env:
+      NXF_ANSI_LOG: false
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+
+      - name: Set up Python
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5
+        with:
+          python-version: "3.11"
+
+      - uses: actions/cache@6849a6489940f00c2f30c0fb92c6274307ccb58a # v4
+        id: cache-pip-pytest
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-pytest
+          restore-keys: |
+            ${{ runner.os }}-pip-pytest
+
+      - name: Install Python dependencies
+        run: python -m pip install --upgrade pip pytest-workflow cryptography
+
+      - uses: actions/setup-java@8df1039502a15bceb9433410b1a100fbe190c53b # v4
+        with:
+          distribution: "temurin"
+          java-version: "17"
+
+      - name: Setup Nextflow
+        uses: nf-core/setup-nextflow@v2
+
+      - name: Setup apptainer
+        if: matrix.profile == 'singularity'
+        uses: eWaterCycle/setup-apptainer@main
+
+      - name: Set up Singularity
+        if: matrix.profile == 'singularity'
+        run: |
+          mkdir -p $NXF_SINGULARITY_CACHEDIR
+          mkdir -p $NXF_SINGULARITY_LIBRARYDIR
+
+      - name: Set up miniconda
+        uses: conda-incubator/setup-miniconda@d2e6a045a86077fb6cad6f5adf368e9076ddaa8d # v3
+        with:
+          miniconda-version: "latest"
+          channels: conda-forge,bioconda
+          python-version: "3.11"
```

Comment:
how about we break out this nf-test setup into a shared workflow, because we have a lot of similar steps in nf-test (but there wer setup miniconda a little bit different for example)

Reply:
Eh, we can but I really hope we'll be completely rid of pytest-workflow soon. Just doesn't seem worth the effort when we already had so much duplication before this anyways.

If you want to address it in another PR go for it!

Reply:
True, fine for now then.